### ISSUE TITLE
Hi, I have some question about Fabricated branch

### ISSUE THREAD

**Issue by bitwangdan** [zhihou7/HOI-CL#2]
In every step training, do you input all the obj embedding and vert  into the .Fabricate model, and then filter out the wrong combination? Or do you only take one obj at a time?

**Comment by zhihou7** [comment]
We first input all the obj embedding to combine with each verb, then remove infeasible HOIs. We think this can balance the object distribution for each verb.

I suggest this implementation of FCL (https://github.com/zhihou7/FCL_VCOCO), which only contains FCL code and is thus clear

**Comment by bitwangdan** [comment]
thank you for your reply, Is the implementation method the same on the V-COCO and HICO data sets?  I am not very familiar with tensorflow, and I have not found the relevant code, on the HICO data set

**Comment by zhihou7** [comment]
Yes, the core part of FCL is similar. But on V-COCO, we do not use verb auxiliary loss since there are only 24 verbs. In this repository (https://github.com/zhihou7/FCL_VCOCO), we include code of FCL as a unique commit: https://github.com/zhihou7/FCL_VCOCO/commit/34ada5eee87389f0560125d403f58bfc96ce5aca. 

Code of FCL on HICO data is https://github.com/zhihou7/HOI-CL/blob/master/lib/networks/Fabricator.py

**Comment by bitwangdan** [comment]
@zhihou7 hi，Can this model be run based on resnet50?  In download_dataset.sh，there is only a  download link for resnet50, but in the FCL project, the model is trained based on resnet101.

**Comment by zhihou7** [comment]
the url (https://drive.google.com/file/d/0B1_fAEgxdnvJR1N3c1FYRGo1S1U/view) in download_dataset.sh is the weights for resnet101.

I also uncomment the line in Train_FCL_HICO.py with resnet50 weights right now. you can update the code. you can change the model name FCL_union_l2_zs_s0_vloss2_varl_gan_dax_rands_rew_aug5_x5new_res101 to FCL_union_l2_zs_s0_vloss2_varl_gan_dax_rands_rew_aug5_x5new_res50 for use resnet50 backbone. However, I do not test resnet50 in this repository. 

regards

**Comment by bitwangdan** [comment]
![image](https://user-images.githubusercontent.com/18326564/119620029-277e4f00-be37-11eb-9f13-e235c1f9cf77.png)
Sorry to bother you again. Is your obj embedding randomly generated? Corresponding to "word2vec_list" in the code, the dimension is verb*obj_num_class*2048

**Comment by zhihou7** [comment]
Yes. You can also use a word embedding to generate it. But, we find the result of randomly initialized embedding is slightly better and simple.
